src/libpgmoneta/wf_link.c

```diff
@@ -94,7 +94,7 @@ link_execute(int server, char* identifier, struct node* i_nodes, struct node** o
    {
       for (int j = number_of_backups - 2; j >= 0 && next_newest == -1; j--)
       {
-         if (backups[j]->valid == VALID_TRUE)
+         if (backups[j]->valid == VALID_TRUE && backups[j]->version == backups[number_of_backups - 1]->version)
```

Comment:
> take 2 backups against a 13 instance, and take 2 backups against a 16 instance using the same server identifier...

Hmm what should be the expected behavior here? Should it be backup_v13_1 <- backup_v13_2 and backup_v16_1 <-backup_v16_2?

> number_of_backups - 1 should likely be j - 1, but latter can't be a negative value...

The logic for this for loop is that the current and newest backup is always the last one (number_of_backups - 1), and we want to link it to the next newest valid backup. And the added condition checks if the next newest backup has the same version, and skip it if it doesn't.

So for example when backup_v16_1 is created, and we want to link it with the next newest, we will first check backup_v13_2, and skip it because the version is different, and so is backup_v13_1. But when backup_v16_2 is created, it will check and see backup_v16_1 is the next newest and of the same version, so it will link with it. 

It is not very easy to install multiple postgres on fedora34. I'll switch to fedora 38 to test it out.

Reply:
> This doesn't look correct - take 2 backups against a 13 instance, and take 2 backups against a 16 instance using the same server identifier...

I got postgres13 and postgres 16 installed on the same fedora39 instance and tested it out. It fits my expectation. V16 backups are not linked with V13 backups. Let me know if I got you wrong.

Reply:
And, the 2 x 13 are linked, and the 2 x 16 are as well ?

Reply:
yep

Reply:
Some proof, the earlier 2 backups are v13, and the later 2 are v16. Cyan files are symlinks. So 20231208070120 is linked to 20231208070118, while 20231208070118 is not linked to 20231208065636
![image](https://github.com/pgmoneta/pgmoneta/assets/89931362/c566d83a-7bb0-415f-8975-029b44848f88)